### PR TITLE
MBL-988: Add GraphQL query FetchBackerProjectsQuery to replace v1 API calls on backer dashboard

### PR DESCRIPTION
**Pull request by amy-at-kickstarter**
# 📲 What

Add a new GraphQL query to power the "backed" and "saved" tabs on the Profile page. This will replace the v1 discover API call on the same page.

# 🤔 Why

- Replace another usage of API v1 in the app
- The 'saved' and 'backed' counts on this page can get out of sync with the actual results list. By porting this to GraphQL, we can include the `totalCount` in the same query, which will let me fix this issue.

# ✅ Acceptance criteria

- This query isn't currently integrated (that will come in a future PR), so the only acceptance criteria is passing tests.

# ⏰ TODO

- `FetchProjectsEnvelope_FetchBackerProjectsQueryDataTests` are currently _not_ passing, because `Project.project(from: fragment, currentUserChosenCurrency: nil)` fails when attempting to parse the test data. I could use some help debugging this - the error I get back is 
```
Could not cast value of type 'NSTaggedPointerString' (0x1bfdf4180) to 'KsApi.GraphAPI.CurrencyCode' (0x11d17b4f0).
```

[I found some code which implies](https://github.com/kickstarter/ios-oss/blob/main/KsApi/mutations/templates/query/FetchAddOnsQueryTemplate.swift#L615) that you need to manually change your test JSON data to be the actual GraphQL types. Is there a way around this, or another solution?

### CODE REVIEW

File: KsApi/queries/templates/FetchBackerProjectsQuery.json
Comment (amy-at-kickstarter) [inline KsApi/queries/templates/FetchBackerProjectsQuery.json:1]: This is data I pulled from staging - do I need to sanitize it at all?

File: KsApi/queries/FetchBackerProjectsQuery.graphql
Comment (amy-at-kickstarter) [inline KsApi/queries/FetchBackerProjectsQuery.graphql:4]: Is there a smaller fragment than `ProjectFragment` I should be using here? This is actually way more data than we need for the page, which is just showing a list of compact cards.

File: Kickstarter.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (amy-at-kickstarter) [inline Kickstarter.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:270]: SPM did this automatically; should I throw it in a separate commit or just let it ride?

File: KsApi/queries/templates/FetchBackerProjectsQueryRequestForTests.graphql
Comment (amy-at-kickstarter) [inline KsApi/queries/templates/FetchBackerProjectsQueryRequestForTests.graphql:1]: This is only added to the test target - it's just here so that the next person who needs to generate one of these queries (which will probably be me!) doesn't have to figure out the full query text again 😁

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| KsApi/MockService.swift:6:1 | warning: (blankLinesAtStartOfScope) Remove leading blank line at the start of a scope. |
| KsApi/MockService.swift:6:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:94:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:186:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:187:1 | warning: (consecutiveBlankLines) Replace consecutive blank lines with a single blank line. |
| KsApi/MockService.swift:371:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:1042:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:1043:1 | warning: (unusedArguments) Mark unused function arguments with _. |
| KsApi/MockService.swift:1043:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/MockService.swift:1044:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/MockService.swift:1044:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/MockService.swift:1049:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:1050:1 | warning: (unusedArguments) Mark unused function arguments with _. |
| KsApi/MockService.swift:1050:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/MockService.swift:1051:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/MockService.swift:1052:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:1052:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/MockService.swift:1057:1 | warning: (consecutiveBlankLines) Replace consecutive blank lines with a single blank line. |
| KsApi/Service.swift:519:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/Service.swift:538:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/Service.swift:539:1 | warning: (blankLinesAtStartOfScope) Remove leading blank line at the start of a scope. |
| KsApi/Service.swift:539:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/Service.swift:540:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/Service.swift:548:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/Service.swift:549:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/Service.swift:555:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/ServiceType.swift:370:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/ServiceType.swift:371:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/ServiceType.swift:372:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/ServiceType.swift:373:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/models/FetchProjectsEnvelope.swift:1:1 | warning: (fileHeader) Use specified source file header template for all files. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryData.swift:1:1 | warning: (fileHeader) Use specified source file header template for all files. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryData.swift:14:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryData.swift:23:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryData.swift:30:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:1:1 | warning: (fileHeader) Use specified source file header template for all files. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:12:1 | warning: (blankLinesAtStartOfScope) Remove leading blank line at the start of a scope. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:14:1 | warning: (blankLinesAtStartOfScope) Remove leading blank line at the start of a scope. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:16:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:17:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:19:1 | warning: (semicolons) Remove semicolons. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:22:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/models/graphql/adapters/FetchProjectsEnvelope+FetchBackerProjectsQueryDataTests.swift:26:1 | warning: (blankLinesAtEndOfScope) Remove trailing blank line at the end of a scope. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:1:1 | warning: (fileHeader) Use specified source file header template for all files. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:1:1 | warning: (sortedImports) Sort import statements alphabetically. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:2:1 | warning: (sortedImports) Sort import statements alphabetically. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:10:1 | warning: (sortedImports) Sort import statements alphabetically. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:15:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:19:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:22:1 | warning: (wrapArguments) Align wrapped function arguments or collection elements. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:25:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:26:1 | warning: (consecutiveBlankLines) Replace consecutive blank lines with a single blank line. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:27:1 | warning: (blankLinesAtStartOfScope) Remove leading blank line at the start of a scope. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:29:1 | warning: (trailingSpace) Remove trailing space at end of a line. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:31:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:34:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:38:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:42:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:45:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/queries/templates/FetchBackerProjectsQueryDataTemplate.swift:46:1 | warning: (blankLinesAtEndOfScope) Remove trailing blank line at the end of a scope. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter.xcodeproj/project.pbxproj
Comment (scottkicks) [inline Kickstarter.xcodeproj/project.pbxproj:7706]: these changes don't seem relevant to this PR. should we revert them?

File: Kickstarter.xcodeproj/project.pbxproj
Comment (scottkicks) [inline Kickstarter.xcodeproj/project.pbxproj:7820]: same question as above. did you mean to remove this?

File: Kickstarter.xcodeproj/project.pbxproj
Comment (scottkicks) [inline Kickstarter.xcodeproj/project.pbxproj:8090]: I wonder what caused these changes? maybe your branch is out of sync with `main`?

**Review by scottkicks — CHANGES_REQUESTED** [review]
Nice work!
Tested  the query on https://staging.kickstarter.com/graphiql and everything looks good!

There are a few of .pbxproj changes I'm not sure we want to have. Can you clarify those for me?

Also, yhanks for catching the rosetta issue in the circleci config

File: Kickstarter.xcodeproj/project.pbxproj
Comment (amy-at-kickstarter) [reply Kickstarter.xcodeproj/project.pbxproj:5506]: Hm, I didn't touch this. Could be Xcode automatically running some cleanup 🤷‍♀️

**Comment by amy-at-kickstarter** [comment]
@scottkicks I went ahead and re-did all my project changes, but XCode still made the same edits.

For example:
```
                                D09362B0225D803600E1411A /* UIViewController+URLTests.swift in Sources */,
                                77319C7E2469B5C70051B755 /* DiscoveryProjectCardViewModelTests.swift in Sources */,
                                A7ED1FDF1E831C5C00BFFA01 /* DiscoveryFiltersViewModelTests.swift in Sources */,
-                               191E60282A953FAB001413B2 /* ProjectTabQuestionAnswerCellViewModelTests.swift in Sources */,
-                               A7ED1FDC1E831C5C00BFFA01 /* DashboardVideoCellViewModelTests.swift in Sources */,
                                8A5CB28424C11819003113D4 /* RewardAddOnSelectionContinueCTAViewModelTests.swift in Sources */,
                                3706409022A9BE9400889CBD /* DateFormatterTests.swift in Sources */,
                                778F891E22D5414F00D095C5 /* Feature+HelpersTests.swift in Sources */,
```

I checked for both `ProjectTabQuestionAnswerCellViewModelTests` and `DashboardVideoCellViewModelTests`, and neither appears to exist any more (in ios-oss or in the project), so my guess is that Xcode is pruning some kind of dead reference.

File: Kickstarter.xcodeproj/project.pbxproj
Comment (scottkicks) [reply Kickstarter.xcodeproj/project.pbxproj:5506]: I hope not. I think this is removing a folder. The DeprecationBanner is related to the old Creator Dashboard stuff that Ingerid just removed.

**Comment by scottkicks** [comment]
>  my guess is that Xcode is pruning some kind of dead reference.

hmm ok. Let's update this branch with `main` and merge once the CI tests pass again. We'll keep an eye on any reference error that might come up

**Review by scottkicks — APPROVED** [review]
LGTM - fyi we typically get 2 approvals before merging.

**Pull request merged**
